packages/webapp/src/components/lists/ReportList/index.tsx

```diff
@@ -58,24 +58,51 @@ export const ReportList: StandardFC<ReportListProps> = wrap(function ReportList(
 			setFilteredData(empty)
 			setCsvFields(empty)
 			setReportType(reportType)
-			setHiddenFields({})
+			setHiddenFields(preferences?.reportList[reportType]?.hiddenFields ?? {})
 			clearFilters()
 		},
-		[setUnfilteredData, setFilteredData, setCsvFields, clearFilters, setHiddenFields, setReportType]
+		[
+			setUnfilteredData,
+			setFilteredData,
+			setCsvFields,
+			clearFilters,
+			setHiddenFields,
+			setReportType,
+			preferences?.reportList
+		]
 	)
 
 	const handleShowFieldsChange = useCallback(
 		(fieldOption: IDropdownOption) => {
+			let _hiddenFields = { ...hiddenFields }
 			if (!fieldOption.selected) {
-				const _hiddenFields = { ...hiddenFields, [fieldOption.key]: true }
-				setHiddenFields(_hiddenFields)
+				_hiddenFields = { ...hiddenFields, [fieldOption.key]: true }
 				clearFilter(fieldOption.key as string)
 			} else {
-				const _hiddenFields = { ...hiddenFields, [fieldOption.key]: undefined }
-				setHiddenFields(_hiddenFields)
+				_hiddenFields = { ...hiddenFields, [fieldOption.key]: undefined }
 			}
+
+			setHiddenFields(_hiddenFields)
+
+			updateUserPreferences(
+				JSON.stringify({
```

Comment:
I think this `JSON.stringify` should be implemented in `updateUserPreferences`, so that we can pass object directly to the method.

Reply:
great point